### PR TITLE
feat: implement multi-track audio player with zero-latency mixing

### LINKED ISSUE #3: Multi-track audio player (player.py)
Tasks for Phase 1 MVP. See PROJECT.md and AGENTS.md for full context.

### PR DESCRIPTION
**Pull request by cyanidesayonara**
## What
Implements MultiTrackPlayer to handle perfectly synced playback of separated stem WAV files.

## How
- Loads all WAV files directly into memory mapping to RAM (~170MB for a 4-minute stereo track).
- Implements a highly optimized sounddevice.OutputStream non-blocking portaudio callback.
- Dynamic matrix summing of stems allows zero-latency mute and solo without allocating memory or pausing the stream.
- Exposes QObject signals (position_changed, state_changed) to drive a UI via a decoupled QTimer.

## Testing
- [x] Full unit test suite passes.
- [x] Manual trigger of portaudio C-callback verifies float32 array accumulation and correct mute/solo precedence without hardware locks.
- [x] seeking and boundary condition constraints tested.

Fixes #3

### CODE REVIEW

File: src/player.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        This runs on a high-priority C thread. It must not block or allocate
+        significant memory.
+        """
+        if status:
+            print(f"Audio callback status: {status}")
+
+        if not self._is_playing or self._current_frame >= self._total_frames:
+            outdata.fill(0.0)
+            raise sd.CallbackStop
+
```
Comment (cyanidesayonara) [inline src/player.py]: `print()` inside a real-time audio callback can block on I/O and cause audible glitches. Consider removing this or gating it behind a debug flag. In production, PortAudio status warnings are rare and not actionable mid-stream.
